### ISSUE TITLE
Infobip react native sdk update Issue

### ISSUE THREAD

**Issue by Rohit-Surawat** [infobip/mobile-messaging-react-native-plugin#33]
Hi All,

We are facing issue related to infobip sdk update, because of that we are facing multiple issue,

Existing Version: 2.2.0
Target Version :4.1.4

Current Implementation:

We have both android sdk and react native plugin in place for info bip notification handling.
1. Infobip there is a module created in android which handles the saving user data, login, logout and handle push notifications.
2. React native side it takes care of deep linking and routing in our application based on the notification type.

Now problem is when i upgrade there is conflict between 5.3.6 android sdk runtime which installed along with react native plugin and android sdk for infobip 4.0.2 

tried upgrading the native sdk to 5.3.6 didn't help throws unable to resolve module.

Any help or guidance will be a great help 
For more info please feel free to comment
Thanks & Regards
Rohit

**Comment by kostap13** [comment]
Hi @Rohit-Surawat 
Could you please provide more details about yours environment?

RN version, node version etc.

**Comment by kostap13** [comment]
And also one other question

> We have both android sdk and react native plugin in place for info bip notification handling.:
Why do need such solution?

Maybe RN plugin 4.1.4 will cover all yours requirements?

**Comment by riskpp** [comment]
Hello @Rohit-Surawat,
Could be that Mobile Messaging react-native plugin can cover all your requirements
- Saving user data, personalize, depersonalize - [Personalization](https://github.com/infobip/mobile-messaging-react-native-plugin/wiki/Users-and-installations#personalization-phone-numbers-emails-and-external-user-id)
- Handling push notifications - "messageReceived", "notificationTapped" events - [Library Events](https://github.com/infobip/mobile-messaging-react-native-plugin/wiki/Library-events)

Usage of the native SDK together with react-native plugin it's rarely seen use case, but we will try to reproduce the issue, could you please provide your versions (react-native version, node version), it'll help us to return with solution faster.

**Comment by Rohit-Surawat** [comment]
Hi Sorry for delay in response ,
Let me provide you detail explanation,

We have native implementation in place for Android OS in our react native application. 

Current Build Gradle Of Android :
org.infobip.mobile.messaging.api:infobip-mobile-messaging-android-sdk:4.0.2@aar

React Native Plugin Version :
  "infobip-mobile-messaging-react-native-plugin": "2.2.0",

Now when i update to latest version
android:
com.infobip:infobip-mobile-messaging-android-sdk:5.3.6@aar

react native version
"infobip-mobile-messaging-react-native-plugin": "4.1.6',

Application is. not starting at all getting following stack trace:
java.lang.RuntimeException: Unable to create application com.emaartechnologies.MainApplication: java.lang.RuntimeException: Requested enabled DevSupportManager, but DevSupportManagerImpl class was not found or could not be created
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5743)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.RuntimeException: Requested enabled DevSupportManager, but DevSupportManagerImpl class was not found or could not be created
        at com.facebook.react.devsupport.b.a(SourceFile:90)
        at com.facebook.s0.p.<init>(SourceFile:245)
        at com.facebook.s0.q.b(SourceFile:287)
        at com.facebook.s0.s.a(SourceFile:89)
        at com.facebook.s0.s.i(SourceFile:39)
        at com.emaartechnologies.MainApplication.onCreate(SourceFile:74)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740)
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6494) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
     Caused by: java.lang.ClassNotFoundException: com.facebook.react.devsupport.DevSupportManagerImpl
        at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:453)
        at java.lang.Class.forName(Class.java:378)
        at com.facebook.react.devsupport.b.a(SourceFile:68)
        at com.facebook.s0.p.<init>(SourceFile:245) 
        at com.facebook.s0.q.b(SourceFile:287) 
        at com.facebook.s0.s.a(SourceFile:89) 
        at com.facebook.s0.s.i(SourceFile:39) 
        at com.emaartechnologies.MainApplication.onCreate(SourceFile:74) 
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740) 
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6494) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.facebook.react.devsupport.DevSupportManagerImpl" on path: DexPathList[[zip file "/data/app/com.emaartechnologies.emaarone.stage-_uAlIrkOr0p9zKDqo3tQ0A==/base.apk"],nativeLibraryDirectories=[/data/app/com.emaartechnologies.emaarone.stage-_uAlIrkOr0p9zKDqo3tQ0A==/lib/x86, /data/app/com.emaartechnologies.emaarone.stage-_uAlIrkOr0p9zKDqo3tQ0A==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)

Already have multidex enabled true for the application also okhttp client is excluded from flipper for api level 21

**Comment by Rohit-Surawat** [comment]
Hi @riskpp @kostap13 Sorry for delay in response ,
Let me provide you detail explanation,

We have native implementation in place for Android OS in our react native application.

Current Build Gradle Of Android :
org.infobip.mobile.messaging.api:infobip-mobile-messaging-android-sdk:4.0.2@aar

React Native Plugin Version :
"infobip-mobile-messaging-react-native-plugin": "2.2.0",

Now when i update to latest version
android:
com.infobip:infobip-mobile-messaging-android-sdk:5.3.6@aar

react native version
"infobip-mobile-messaging-react-native-plugin": "4.1.6',

Application is. not starting at all getting following stack trace:
java.lang.RuntimeException: Unable to create application
java.lang.RuntimeException: Requested enabled DevSupportManager, but DevSupportManagerImpl class was not found or could not be created
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5743)
at android.app.ActivityThread.-wrap1(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6494)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
Caused by: java.lang.RuntimeException: Requested enabled DevSupportManager, but DevSupportManagerImpl class was not found or could not be created
at com.facebook.react.devsupport.b.a(SourceFile:90)
at com.facebook.s0.p.(SourceFile:245)
at com.facebook.s0.q.b(SourceFile:287)
at com.facebook.s0.s.a(SourceFile:89)
at com.facebook.s0.s.i(SourceFile:39)
at MainApplication.onCreate(SourceFile:74)
at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119)
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740)
at android.app.ActivityThread.-wrap1(Unknown Source:0) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loop(Looper.java:164) 
at android.app.ActivityThread.main(ActivityThread.java:6494) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
Caused by: java.lang.ClassNotFoundException: com.facebook.react.devsupport.DevSupportManagerImpl
at java.lang.Class.classForName(Native Method)
at java.lang.Class.forName(Class.java:453)
at java.lang.Class.forName(Class.java:378)
at com.facebook.react.devsupport.b.a(SourceFile:68)
at com.facebook.s0.p.(SourceFile:245) 
at com.facebook.s0.q.b(SourceFile:287) 
at com.facebook.s0.s.a(SourceFile:89) 
at com.facebook.s0.s.i(SourceFile:39) 
at MainApplication.onCreate(SourceFile:74) 
at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119) 
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740) 
at android.app.ActivityThread.-wrap1(Unknown Source:0) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loop(Looper.java:164) 
at android.app.ActivityThread.main(ActivityThread.java:6494) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.facebook.react.devsupport.DevSupportManagerImpl" on path: DexPathList[[zip file "/data/a
at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
at java.lang.ClassLoader.loadClass(ClassLoader.java:312)

Already have multidex enabled true for the application also okhttp client is excluded from flipper for api level 21

**Comment by riskpp** [comment]
Hi @Rohit-Surawat,
Could you please check that you can't use just react-native plugin, without native sdk? As I [mentioned above](https://github.com/infobip/mobile-messaging-react-native-plugin/issues/33#issuecomment-1009659727) plugin has all features, which exist in native sdk.

Which react-native version is used in your project? could you please try to upgrade to latest supported by our plugin - 0.66.3

BR, Olga

**Comment by Rohit-Surawat** [comment]
Current React Native Version : 0.65.1

**Comment by Rohit-Surawat** [comment]
Full Package JSON Details: I have doubt if updating info bip is causing conflict with one of below mentioned libs.
"@amplitude/react-native": "^2.5.0",
    "@ebizon/react-native-advance-draggable-view": "^1.0.13",
    "@gorhom/bottom-sheet": "^2",
    "@hmscore/react-native-hms-map": "^5.0.1-300",
    "@jumpn/react-native-jetifier": "^0.1.4",
    "@monterosa/react-native-parallax-scroll": "^1.8.0",
    "@react-native-community/blur": "^3.6.0",
    "@react-native-community/cameraroll": "^4.0.4",
    "@react-native-community/datetimepicker": "^2.1.0",
    "@react-native-community/image-editor": "^2.2.0",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "^4.7.0",
    "@react-native-community/progress-bar-android": "^1.0.4",
    "@react-native-community/progress-view": "^1.3.1",
    "@react-native-firebase/analytics": "^7.6.9",
    "@react-native-firebase/app": "^8.4.7",
    "@react-native-firebase/crashlytics": "^10.1.0",
    "@twotalltotems/react-native-otp-input": "^1.3.11",
    "aes-js": "^3.1.2",
    "axios": "^0.18.0",
    "buffer": "^5.2.1",
    "crypto-js": "^3.1.9-1",
    "d3": "^5.9.2",
    "fast-xml-parser": "^3.17.5",
    "google-libphonenumber": "^3.2.2",
    "i18n-js": "^3.3.0",
    "immutable": "^4.0.0-rc.12",
    "infobip-mobile-messaging-react-native-plugin": "4.1.6",
    "jail-monkey": "^2.2.0",
    "libphonenumber-js": "^1.7.26",
    "lodash": "^4.17.11",
    "lodash.memoize": "^4.1.2",
    "lottie-ios": "3.1.8",
    "lottie-react-native": "^3.5.0",
    "metro-react-native-babel-transformer": "^0.64.0",
    "moment": "^2.24.0",
    "moment-timezone": "^0.5.34",
    "native-base": "^2.13.8",
    "pkcs7": "^1.0.2",
    "prop-types": "^15.7.2",
    "react": "17.0.2",
    "react-native": "0.66.3",
    "react-native-actionsheet": "^2.4.2",
    "react-native-advance-image-cropper": "^1.1.0",
    "react-native-audio": "^4.3.0",
    "react-native-awesome-alerts": "^1.3.2",
    "react-native-blob-util": "^0.13.16",
    "react-native-calendar-events": "^2.2.0",
    "react-native-calendars": "1.829.0",
    "react-native-camera": "3.26.0",
    "react-native-camera-roll-picker": "^2.0.0",
    "react-native-carrier-info": "^1.1.2",
    "react-native-code-push": "^7.0.4",
    "react-native-config": "^0.12.0",
    "react-native-contacts": "^5.2.2",
    "react-native-datepicker": "^1.7.2",
    "react-native-device-info": "^2.1.2",
    "react-native-dimension": "^1.0.6",
    "react-native-document-picker": "3.5.1",
    "react-native-fast-image": "^8.3.7",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.6.0",
    "react-native-image-crop-picker": "^0.28.0",
    "react-native-image-placeholder": "^1.0.14",
    "react-native-image-resizer": "^1.2.3",
    "react-native-image-rotate": "^2.1.0",
    "react-native-image-to-pdf": "^1.2.0",
    "react-native-in-app-review": "^3.2.2",
    "react-native-keyboard-aware-scroll-view": "github:APSL/react-native-keyboard-aware-scroll-view#pull/501/head",
    "react-native-keychain": "^8.0.0",
    "react-native-linear-gradient": "^2.5.6",
    "react-native-localization": "^2.1.7",
    "react-native-localize": "^1.3.0",
    "react-native-maps": "^0.28.0",
    "react-native-mime-types": "^2.2.1",
    "react-native-modal": "^13.0.0",
    "react-native-modal-popover": "^0.0.12",
    "react-native-pager-view": "^5.4.8",
    "react-native-parsed-text": "^0.0.21",
    "react-native-payments": "^0.8.3",
    "react-native-pdf": "^6.3.0",
    "react-native-permissions": "3.0.5",
    "react-native-phone-call": "^1.0.9",
    "react-native-phone-input": "^0.2.1",
    "react-native-pinning-ssl": "^1.0.3",
    "react-native-popup-menu": "^0.15.6",
    "react-native-reanimated": "^1.7.0",
    "react-native-render-html": "^5.1.1",
    "react-native-responsive-fontsize": "^0.5.0",
    "react-native-restart": "^0.0.17",
    "react-native-root-toast": "^3.0.2",
    "react-native-safe-area": "^0.5.1",
    "react-native-safe-area-context": "^3.3.2",
    "react-native-screens": "^2.17.0",
    "react-native-share": "^1.2.1",
    "react-native-signature-capture": "^0.4.10",
    "react-native-skeleton-placeholder": "^4.0.0",
    "react-native-snackbar": "^2.4.0",
    "react-native-snap-carousel": "^3.7.5",
    "react-native-sound": "^0.11.0",
    "react-native-star-rating": "^1.1.0",
    "react-native-svg": "^9.4.0",
    "react-native-svg-charts": "^5.2.0",
    "react-native-tab-view": "^3.1.1",
    "react-native-thumbnail-video": "^0.1.2",
    "react-native-touch-id": "^4.4.1",
    "react-native-version-number": "^0.3.6",
    "react-native-video": "^5.0.2",
    "react-native-webview": "^11.14.2",
    "react-native-xml2js": "^1.0.3",
    "react-native-youtube": "^2.0.0",
    "react-native-youtube-iframe": "^1.3.0",
    "react-navigation": "^3.5.1",
    "react-number-format": "^4.6.4",
    "react-redux": "^7.2.0",
    "redux": "^4.0.0",
    "redux-devtools-extension": "^2.13.8",
    "redux-immutable": "^4.0.0",
    "redux-saga": "^1.0.2",
    "reselect": "^4.0.0",
    "rn-fetch-blob": "^0.12.0",
    "search-params": "^2.1.3",
    "uuid": "^3.3.2"

 "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "babel-eslint": "^10.0.1",
    "babel-generator": "^7.0.0-beta.3",
    "babel-jest": "^26.6.3",
    "babel-plugin-transform-remove-console": "^6.9.4",
    "babel-preset-react-native": "^5.0.2",
    "eslint": "7.14.0",
    "eslint-config-airbnb": "18.1.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-import": "^2.20.1",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-prettier": "^3.1.3",
    "eslint-plugin-react": "^7.19.0",
    "eslint-plugin-react-hooks": "2.5.0",
    "eslint-plugin-react-native": "^3.6.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.66.0",
    "react-native-codegen": "^0.0.7",
    "react-native-obfuscating-transformer": "1.0.0",
    "react-test-renderer": "^17.0.2"
  },

**Comment by agupta-avenir** [comment]
@riskpp  Any update on this issue. We are struggling to update the infobip version in our App from 2.2.0 -> 4.1.6
This is some more details of the error:
ERROR  Invariant Violation: "appName" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called.

**Comment by agupta-avenir** [comment]
@riskpp  Just an update we were successfully able to update to 4.0.0 but facing issue with v4.1.6 update

**Comment by agupta-avenir** [comment]
@kas-kad If you can throw some light on this issue as well will be great.

**Comment by riskpp** [comment]
Hi @agupta-avenir, sorry for delay in the answer
Could you please send your proguard rules file, could be somehow related with the issue..
I've check for the following environment:
- mobile-messaging plugin version 4.1.6
- react-native: 0.65.1
- node: 17.0.1
that simple application created by command `react-native init TestRN65_1 --version 0.65.1` is working fine on Android.

**Comment by agupta-avenir** [comment]
Here it is:

`# Copyright (c) Facebook, Inc. and its affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.

# Add project specific ProGuard rules here.
# By default, the flags in this file are appended to flags specified
# in /usr/local/Cellar/android-sdk/24.3.3/tools/proguard/proguard-android.txt
# You can edit the include path and order by changing the proguardFiles
# directive in build.gradle.
#
# For more details, see
#   https://developer.android.com/guide/developing/tools/proguard.html

# Add any project specific keep options here:

# Play Core Proguard Rules: Play In-app Review Start
-keep class com.google.android.play.core.review.ReviewManager {
  public com.google.android.play.core.tasks.Task requestReviewFlow();
  public com.google.android.play.core.tasks.Task launchReviewFlow(android.app.Activity, com.google.android.play.core.review.ReviewInfo);
}

-keepnames class com.google.android.play.core.review.ReviewInfo

-keep class com.google.android.play.core.review.ReviewManagerFactory {
  <init>();

  public static com.google.android.play.core.review.ReviewManager create(android.content.Context);
}

-keep class com.google.android.play.core.review.testing.FakeReviewManager {
  public <init>(android.content.Context);
  public com.google.android.play.core.tasks.Task requestReviewFlow();
  public com.google.android.play.core.tasks.Task launchReviewFlow(android.app.Activity, com.google.android.play.core.review.ReviewInfo);
}
# Play Core Proguard Rules: Play In-app Review End

#-keep class org.infobip.reactlibrary.mobilemessaging.Configuration* { *; }
#-keep class org.infobip.reactlibrary.mobilemessaging.CacheManager* { *; }

-keep class com.huawei.**{*;}

# Disabling obfuscation is useful if you collect stack traces from production crashes
# (unless you are using a system that supports de-obfuscate the stack traces).
# -dontobfuscate

# React Native

# Keep our interfaces so they can be used by other ProGuard rules.
# See https://sourceforge.net/p/proguard/bugs/466/
-keep,allowobfuscation @interface com.facebook.proguard.annotations.DoNotStrip
-keep,allowobfuscation @interface com.facebook.proguard.annotations.KeepGettersAndSetters
-keep,allowobfuscation @interface com.facebook.common.internal.DoNotStrip

# Do not strip any method/class that is annotated with @DoNotStrip
-keep @com.facebook.proguard.annotations.DoNotStrip class *
-keep @com.facebook.common.internal.DoNotStrip class *
-keepclassmembers class * {
    @com.facebook.proguard.annotations.DoNotStrip *;
    @com.facebook.common.internal.DoNotStrip *;
}

-keepclassmembers @com.facebook.proguard.annotations.KeepGettersAndSetters class * {
  void set*(***);
  *** get*();
}

-keep class * extends com.facebook.react.bridge.JavaScriptModule { *; }
-keep class * extends com.facebook.react.bridge.NativeModule { *; }
-keepclassmembers,includedescriptorclasses class * { native <methods>; }
-keepclassmembers class *  { @com.facebook.react.uimanager.annotations.ReactProp <methods>; }
-keepclassmembers class *  { @com.facebook.react.uimanager.annotations.ReactPropGroup <methods>; }

-dontwarn com.facebook.react.**
-keep,includedescriptorclasses class com.facebook.react.bridge.** { *; }

# okhttp

-keepattributes Signature
-keepattributes *Annotation*
-keep class okhttp3.** { *; }
-keep interface okhttp3.** { *; }
-dontwarn okhttp3.**

-keep class com.facebook.react.devsupport.** { *; }
-dontwarn com.facebook.react.devsupport.**
# okio
-keep class com.mypackage.BuildConfig { *; }

-keep class sun.misc.Unsafe { *; }
-dontwarn java.nio.file.*
-dontwarn org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
-dontwarn okio.**
-keep class io.invertase.firebase.** { *; }
-dontwarn io.invertase.firebase.**
-keep class com.facebook.jni.** { *; }
-ignorewarnings
-keepattributes *Annotation*
-keepattributes Exceptions
-keepattributes InnerClasses
-keepattributes SourceFile,LineNumberTable
-keep class com.hianalytics.android.**{*;}
-keep class com.huawei.updatesdk.**{*;}
-keep class com.huawei.hms.**{*;}`

**Comment by riskpp** [comment]
With provided proguard rules I also got error from this [comment](https://github.com/infobip/mobile-messaging-react-native-plugin/issues/33#issuecomment-1016425452), but managed to solve it using [this solution](https://github.com/facebook/react-native/issues/9043#issuecomment-298832659). Looks not related to mobile-messaging plugin.

For mobile-messaging plugin in pro-guard only this is needed:
```
# Application classes that will be serialized/deserialized over Gson
-keep class org.infobip.reactlibrary.mobilemessaging.Configuration* { *; }
-keep class org.infobip.reactlibrary.mobilemessaging.CacheManager* { *; }
```

**Comment by riskpp** [comment]
@agupta-avenir I see in provided proguard rules:
```
-keep class com.mypackage.BuildConfig { *; }
```
Do you have package named "com.mypackage"? It should be changed to your package name. "ReactConfig: Could not find BuildConfig class" issue could be related to this.

**Comment by agupta-avenir** [comment]
@riskpp  Not sure if this is impacting but til version 4.0.0 its working with the same code base.

**Comment by riskpp** [comment]
Hi @agupta-avenir, could you please try to change in your proguard rules package name to real one?
Would be great to have minimal example with which we can reproduce the issue, note that we already tried minimal react-native application with your environment and unfortunately can't reproduce the issue.

BR, Olga

**Comment by agupta-avenir** [comment]
@riskpp  Finally able to fix the issue. Yes it was due to proguard config. Thanks a lot for help and support.
We just got into another issue after updating the sdk where on iOS debugging has stopped working. Any help on that would be great.

**Comment by riskpp** [comment]
Hi @agupta-avenir , great that you managed to solve the initial problem!
Could you please clarify, which debugging doesn't work on iOS? Do you mean flipper or debugging through xCode?

**Comment by imkrishh** [comment]
@riskpp with infobip `4.1.6` version the Initialize method (`mobileMessaging.init`)  in iOS is crashing in chrome debug mode.

React Native Version  `0.65.1` 
MobileMessaging Version `9.2.11` (was 9.1.10)
infobip-mobile-messaging-react-native-plugin `4.1.6` (was 4.0.0)

![Simulator Screen Shot infobip init crash](https://user-images.githubusercontent.com/48957457/154003429-2cafabef-b3b3-47da-b2f1-6010ab3b8bfc.png)

**Comment by riskpp** [comment]
Hi @imkrishh, we started working on it, will inform you about fix

**Comment by riskpp** [comment]
Should be fixed with new version - [5.0.1](https://github.com/infobip/mobile-messaging-react-native-plugin/releases/tag/5.0.1)

**Comment by riskpp** [comment]
Was fixes in latest version
